DigitalLearningSolutions.Web/Controllers/RegisterController.cs

```diff
@@ -0,0 +1,169 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers
+{
+    using System;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Extensions;
+    using DigitalLearningSolutions.Web.Models;
+    using DigitalLearningSolutions.Web.ServiceFilter;
+    using DigitalLearningSolutions.Web.ViewModels.Register;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    public class RegisterController : Controller
+    {
+        private const string CookieName = "RegistrationData";
+        private readonly ICentresService centresService;
+        private readonly IJobGroupsService jobGroupsService;
+
+        public RegisterController(ICentresService centresService, IJobGroupsService jobGroupsService)
+        {
+            this.centresService = centresService;
+            this.jobGroupsService = jobGroupsService;
+        }
+
+        public IActionResult Index()
+        {
+            if (User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home");
+            }
+
+            var delegateRegistrationData = TempData.Get<DelegateRegistrationData>();
+
+            if (delegateRegistrationData == null || !Request.Cookies.ContainsKey(CookieName))
+            {
+                var id = Guid.NewGuid();
+                delegateRegistrationData = new DelegateRegistrationData { Id = id };

```

Comment:
Should we either:
1. Use the id created in the DelegateRegistrationData constructor 
2. Use the id generated in code, but add the id as a param in the DelegateRegistrationData constructor so we don't need to set the id and then override its value

Reply:
Good idea, I'll just use the one from the constructor.